### PR TITLE
MINOR Moved a few log segment util methods from LocalLog to LogFileUtils

### PR DESCRIPTION
**Pull request by satishd**
MINOR Moved a few log segment util methods from LocalLog to LogFileUtils.

This is related to the changes required for [KAFKA-14481](https://issues.apache.org/jira/browse/KAFKA-14481)

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

File: storage/src/main/java/org/apache/kafka/storage/internals/log/LogFileUtils.java [hunk 1]
```diff
@@ -72,4 +92,99 @@ private static String filenamePrefixFromOffset(long offset) {
         return nf.format(offset);
     }
 
+    /**
+     * Construct a log file name in the given dir with the given base offset and the given suffix
```
Comment (showuon) [inline storage/src/main/java/org/apache/kafka/storage/internals/log/LogFileUtils.java]: This constructor doesn't provide `suffix`. Please update the java doc.

File: storage/src/main/java/org/apache/kafka/storage/internals/log/LogFileUtils.java [hunk 2]
```diff
@@ -72,4 +92,99 @@ private static String filenamePrefixFromOffset(long offset) {
...
+     * @param dir    The directory in which the log will reside
+     * @param offset The base offset of the log file
+     * @param suffix The suffix to be appended to the file name (e.g. "", ".deleted", ".cleaned", ".swap", etc.)
+     */
+    public static File logFile(File dir, long offset, String suffix) {
+        return new File(dir, filenamePrefixFromOffset(offset) + LOG_FILE_SUFFIX + suffix);
+    }
+
+    /**
+     * Construct an index file name in the given dir using the given base offset and the given suffix
```
Comment (showuon) [inline storage/src/main/java/org/apache/kafka/storage/internals/log/LogFileUtils.java]: ditto, here and below

File: storage/src/main/java/org/apache/kafka/storage/internals/log/LogFileUtils.java [hunk 3]
```diff
@@ -72,4 +92,99 @@ private static String filenamePrefixFromOffset(long offset) {
...
+     * @param dir    The directory in which the log will reside
+     * @param offset The base offset of the log file
+     * @param suffix The suffix to be appended to the file name ("", ".deleted", ".cleaned", ".swap", etc.)
+     */
+    public static File transactionIndexFile(File dir, long offset, String suffix) {
+        return new File(dir, filenamePrefixFromOffset(offset) + TXN_INDEX_FILE_SUFFIX + suffix);
+    }
+
+    /**
+     * Returns the offset for the given file. The file name is of the form: {number}.{suffix}. This method extracts
```
Comment (showuon) [inline storage/src/main/java/org/apache/kafka/storage/internals/log/LogFileUtils.java]: nit: Returns the offset for the given file. -> Return the offset from the given file.

**Review by showuon — COMMENTED** [review]
Thanks for the patch. Left some comments.

**Comment by satishd** [comment]
Thanks @showuon for the review. Addressed them with the latest commit.

File: core/src/main/scala/kafka/log/LocalLog.scala [hunk 4]
```diff
@@ -686,40 +685,6 @@ object LocalLog extends Logging {
...
 
-  /**
-   * Construct an index file name in the given dir using the given base offset and the given suffix
-   *
-   * @param dir The directory in which the log will reside
-   * @param offset The base offset of the log file
-   * @param suffix The suffix to be appended to the file name ("", ".deleted", ".cleaned", ".swap", etc.)
-   */
-  private[log] def offsetIndexFile(dir: File, offset: Long, suffix: String = ""): File =
-    new File(dir, filenamePrefixFromOffset(offset) + IndexFileSuffix + suffix)
```
Comment (junrao) [inline core/src/main/scala/kafka/log/LocalLog.scala:697]: Should we remove IndexFileSuffix in LocalLog now that it's moved to LogFileUtils?

File: storage/src/main/java/org/apache/kafka/storage/internals/log/LogFileUtils.java [hunk 5]
```diff
@@ -72,4 +92,99 @@ private static String filenamePrefixFromOffset(long offset) {
         return nf.format(offset);
     }
 
+    /**
+     * Construct a log file name in the given dir with the given base offset.
+     *
+     * @param dir    The directory in which the log will reside
+     * @param offset The base offset of the log file
+     */
+    public static File logFile(File dir, long offset) {
```
Comment (junrao) [inline storage/src/main/java/org/apache/kafka/storage/internals/log/LogFileUtils.java:101]: Should we remove `logFile` in LocalLog?

File: storage/src/main/java/org/apache/kafka/storage/internals/log/LogFileUtils.java [hunk 6]
```diff
@@ -72,4 +92,99 @@ private static String filenamePrefixFromOffset(long offset) {
...
+    }
+
+    /**
+     * Returns the offset from the given file. The file name is of the form: {number}.{suffix}. This method extracts
+     * the number from the given file's name.
+     *
+     * @param file file with the offset information as part of its name.
+     * @return offset of the given file
+     */
+    public static Long offsetFromFile(File file) {
```
Comment (junrao) [inline storage/src/main/java/org/apache/kafka/storage/internals/log/LogFileUtils.java:186]: Should we remove `offsetFromFile` from LocalLog?

**Review by junrao — COMMENTED** [review]
@satishd : Thanks for the PR. A few comments below.

File: storage/src/main/java/org/apache/kafka/storage/internals/log/LogFileUtils.java (hunk 6 above)
Comment (satishd) [reply storage/src/main/java/org/apache/kafka/storage/internals/log/LogFileUtils.java:186]: This is already removed from LocalLog. Are you asking whether to remove this method from here and use `offsetFromFileName` instead?

**Comment by satishd** [comment]
Thanks @junrao for the review. Addressed them with the latest commit/comments.

**Review by junrao — APPROVED** [review]
@satishd : Thanks for the updated PR. LGTM

**Review by showuon — APPROVED** [review]

**Comment by satishd** [comment]
Test failures do not seem to be related to this change, merging these changes to trunk.

**Comment by ijuma** [comment]
I didn't see this until now. I actually think the right approach is to push some of these methods to the respective classes (eg if it's related to an index, the method should be in the index class). I starting working on a PR for that and will submit it later.

**Comment by ijuma** [comment]
@satishd How is this related to moving `LogSegment`? It doesn't seem like that was updated at all?

**Comment by satishd** [comment]
> @satishd How is this related to moving `LogSegment`? It doesn't seem like that was updated at all?

Methods like the below have usages in `LogSegment` class. Those will be updated as part of [KAFKA-14481](https://issues.apache.org/jira/browse/KAFKA-14481) changes. 

```
UnifiedLog.offsetIndexFile(dir, baseOffset, fileSuffix)
UnifiedLog.timeIndexFile(dir, baseOffset, fileSuffix)
UnifiedLog.transactionIndexFile(dir, baseOffset, fileSuffix)
UnifiedLog.logFile(dir, baseOffset, fileSuffix)
```

**Pull request merged**